library/general/src/lib/yast2/popup.rb

```diff
@@ -0,0 +1,318 @@
+      # @return [Symbol] symbol of pressed button. If timeout appear,
+      #   then button set in focus parameter is returned. If user click on 'x' button in window
+      #   then `:cancel` symbol is returned.
+      #
+      # @example pair of old and new API calls
+      #   Yast::Popup.Message(text)
+      #   Yast2::Popup.show(text)
+      #
+      #   Yast::Popup.MessageDetails(text, details)
+      #   Yast2::Popup.show(text, details: details)
+      #
+      #   Yast::Popup.TimedError(text, seconds)
+      #   Yast2::Popup.show(text, headline: :error, timeout: seconds)
+      #
+      #   Yast::Popup.TimedErrorAnyQuestion(headline, message, yes_button_message, no_button_message, focus, timeout_seconds)
+      #   Yast2::Popup.show(message, headline: headline, timeout: timeout_seconds, buttons: { yes: yes_button_message, no: no_button_message), focus: :yes)
+
+      #   Yast::Popup.TimedLongNotify(message, timeout_seconds)
+      #   Yast2::Popup.show(message, richtext: true, timeout: timeout_seconds)
+      def show(message, details: "", headline: "", timeout: 0, focus: nil, buttons: :ok, richtext: :auto, style: :notice)
+        textdomain "base"
+        buttons = generate_buttons(buttons)
+        headline = generate_headline(headline)
+        # add default focus button before adding details, as details should not be focussed
+        focus = buttons.keys.first if focus.nil?
+        buttons = add_details_button(buttons) unless details.empty?
+        buttons = add_stop_button(buttons) if timeout > 0
+        check_arguments!(message, details, timeout, focus, buttons)
+        content_res = content(body(headline, message, richtext, timeout), buttons)
+
+        event_loop(content_res, focus, timeout, details, style)
+      end
+
+      # Shows feedback till given block does not finish.
+      # @param message [String] message to show
+      # @param headline [String] sets popup headline. String is shown.
+      #   If empty string is passed no headline is shown.
+      # @yield [] operation that needs to show feedback
+      # @return result of block
+      def feedback(message, headline: "", &block)
+        headline = generate_headline(headline)
+        if !message.is_a?(::String)
+          raise ArgumentError, "Invalid value #{message.inspect} of parameter message"
+        end
+
+        body = VBox(
+          VSpacing(0.4),
+          *headline_widgets(headline),
+          Left(Label(message))
+        )
+        content_res = content(body, {})
+
+        res = Yast::UI.OpenDialog(content_res)
+        raise "Failed to open dialog, see logs." unless res
+        begin
+          block.call
+        ensure
+          Yast::UI.CloseDialog
+        end
+      end
+
+    private
+
+      include Yast::UIShortcuts
+
+      def check_arguments!(message, details, timeout, focus, buttons)
+        if !message.is_a?(::String)
+          raise ArgumentError, "Invalid value #{message.inspect} of parameter message"
```

Comment:
NP: I think it would be nice to include the expected type in the error messages, like for the buttons.

Reply:
buttons does not have it specified. Only focus prints list of buttons that can be focused. I don't want to make exception too long and print mainly the most important data, if user see it, so it is less prone for error when reporting or screenshooting it.